### ISSUE TITLE
Console auto completion for arguments with default value

### ISSUE THREAD

**Issue by dmolineus** [symfony/symfony#44505]
### Symfony version(s) affected

5.4|6.x

### Description

The new console auto completion feature is awesome but unfortunately doesn't work quite well with arguments defining default values.

Given is a console command with two argument `command [<task=default> [<dir>]]` where the first argument `task` has a default value. 

*Expected behaviour:* If I enter `command <TAB>` I expect to the the suggested values for `task`. Instead I see the suggestions for `dir`.

### How to reproduce

- Create a console command with 2 arguments
- The first argument has a default value
- Enable the auto completion and provide suggestions for these two arguments

### Possible Solution

Default values has to be ignored then choosing the current argument for suggestions

**Comment by GromNaN** [comment]
I confirm the bug.

Given this command class:
```php
#[AsCommand(name: 'demo')]
class DemoCommand extends Command
{
    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
    {
        if ($input->mustSuggestArgumentValuesFor('task')) {
            $suggestions->suggestValues(['task1', 'task2']);
        } elseif ($input->mustSuggestArgumentValuesFor('dir')) {
            $suggestions->suggestValues(['dir1', 'dir2']);
        }
    }

    protected function configure(): void
    {
        $this
            ->addArgument('task', InputArgument::OPTIONAL,'1st arg', 'default')
            ->addArgument('dir', InputArgument::OPTIONAL,'2nd arg');
    }
}
```

When I hit `bin/console demo |`, I get the suggestions for the 2nd argument (incorrect).

```
Input: ("|" indicates the cursor position)
  bin/console demo |
Command:
  bin/console _complete -sbash -c2 -S5.4.1-DEV -ibin/console -idemo
Messages:
  Completing using the App\Command\DemoCommand class.
  Completing argument_value for dir
  Current value: 
Suggestions:
  dir1 dir2

```

Then `bin/console demo dir|` suggestion values for the 1st argument (correct)
```
Input: ("|" indicates the cursor position)
  bin/console demo dir|
Command:
  bin/console _complete -sbash -c2 -S5.4.1-DEV -ibin/console -idemo -idir
Messages:
  Completing using the App\Command\DemoCommand class.
  Completing argument_value for task
  Current value: dir
Suggestions:
  task1 task2
```

**Comment by wouterj** [comment]
In case someone wants to look at this, this is probably an issue in `CompletionInput`. You can check this by adding a (hopefully failing) test to `Symfony\Component\Console\Tests\Completion\CompletionInputTest`.

**Comment by GromNaN** [comment]
I'm looking.

**Comment by GromNaN** [comment]
The `help` command is affected by this bug too. Here is the fix: #44730
